packages/dictd/dictd.patch

```diff
@@ -0,0 +1,108 @@
+ int pidFile_set; /* 1 if set by command line option */
+ 
+ const char         *daemon_service     = DICT_DEFAULT_SERVICE;
+diff -u -r ./net.c ../dictd-1.12.1.orig/net.c
+--- ./net.c	2018-02-07 15:32:37.016533093 +0900
++++ ../dictd-1.12.1.orig/net.c	2010-08-22 02:55:40.000000000 +0900
+@@ -65,9 +65,7 @@
+ {
+    struct hostent     *hostEntry;
+    struct servent     *serviceEntry;
+-   /*
+    struct protoent    *protocolEntry;
+-   */
+    struct sockaddr_in ssin;
+    int                s;
+    int                hosts = 0;
+@@ -81,10 +79,8 @@
+    } else if (!(ssin.sin_port = htons(atoi(service))))
+       return NET_NOSERVICE;
+ 
+-   /*
+    if (!(protocolEntry = getprotobyname("tcp")))
+       return NET_NOPROTOCOL;
+-   */
+    
+    if ((hostEntry = gethostbyname(host))) {
+       ++hosts;
+@@ -96,7 +92,7 @@
+ 	 memcpy( &ssin.sin_addr.s_addr, *current, hostEntry->h_length );
+ 	 PRINTF(DBG_VERBOSE,
+ 		("Trying %s (%s)\n",host,inet_ntoa(ssin.sin_addr)));
+-	 if ((s = socket(PF_INET, SOCK_STREAM, 6)) < 0)
++	 if ((s = socket(PF_INET, SOCK_STREAM, protocolEntry->p_proto)) < 0)
+ 	    err_fatal_errno( __func__, "Can't open socket on port %d\n",
+ 			     ntohs(ssin.sin_port) );      
+ 	 if (connect(s, (struct sockaddr *)&ssin, sizeof(ssin)) >= 0)
+@@ -104,7 +100,7 @@
+ 	 close(s);
+       }
+    } else {
+-      if ((s = socket(PF_INET, SOCK_STREAM, 6)) < 0)
++      if ((s = socket(PF_INET, SOCK_STREAM, protocolEntry->p_proto)) < 0)
+ 	 err_fatal_errno( __func__, "Can't open socket on port %d\n",
+ 			  ntohs(ssin.sin_port) );
+       if (connect(s, (struct sockaddr *)&ssin, sizeof(ssin)) >= 0)
+@@ -121,9 +117,7 @@
+    int queueLength)
+ {
+    struct servent     *serviceEntry;
+-   /*
+    struct protoent    *protocolEntry;
+-   */
+    struct sockaddr_in ssin;
+    int                s;
+    const int          one = 1;
+@@ -137,12 +131,10 @@
+    } else if (!(ssin.sin_port = htons(atoi(service))))
+       err_fatal( __func__, "Can't get \"%s\" service entry\n", service );
+ 
+-   /*
+    if (!(protocolEntry = getprotobyname("tcp")))
+       err_fatal( __func__, "Can't get \"tcp\" protocol entry\n" );
+-   */
+    
+-   if ((s = socket(PF_INET, SOCK_STREAM, 6)) < 0)
++   if ((s = socket(PF_INET, SOCK_STREAM, protocolEntry->p_proto)) < 0)
+       err_fatal_errno( __func__, "Can't open socket on port %d\n",
+ 		       ntohs(ssin.sin_port) );
+ 
+diff -u -r ./parse.c ../dictd-1.12.1.orig/parse.c
+--- ./parse.c	2018-02-07 15:32:52.416533087 +0900
++++ ../dictd-1.12.1.orig/parse.c	2008-12-08 01:50:05.000000000 +0900
+@@ -90,10 +90,10 @@
+    char              *buffer;
+    const char        **pt;
+    static const char *cpp = NULL;
+-   static const char *cpps[] = { "@TERMUX_PREFIX@/lib/cpp",
+-                                 "@TERMUX_PREFIX@/usr/lib/cpp",
+-                                 "@TERMUX_PREFIX@/usr/ccs/lib/cpp",	/* Solaris */
+-                                 "@TERMUX_PREFIX@/usr/lang/cpp",
```

Comment:
one /usr/ too many here, TERMUX_PREFIX=/data/data/com.termux/files/usr

Reply:
Thanks, I will fix it.